web/interceptor/access_control.go

```diff
@@ -0,0 +1,150 @@
+package interceptor
+
+import (
+	"context"
+	"strings"
+
+	"github.com/quickfeed/quickfeed/qf"
+	"github.com/quickfeed/quickfeed/web/auth"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+type (
+	role      int
+	roles     []role
+	requestID interface {
+		FetchID(string) uint64
+	}
+)
+
+const (
+	// user role implies that user attempts to access information about himself.
+	user role = iota
```

Comment:
I've read somewhere (can't remember where), that we should have a `none` value as the first entry for enum/iota types. This makes sense also because the database is not great when it comes to updating things to the zero value etc...

That is make sense here?

Reply:
Maybe it could serve as the `everyone` else role.

Reply:
Made it a `none` value, or should it have some other name?

Reply:
Sounds good to me!